### PR TITLE
Flat transaction

### LINKED ISSUE #828: Revisit "inner" transaction in transaction model
Having an "inner" transaction inside the transaction model seems to be confusing and redundant:

```
tx = {
    'id': '…',
    'transaction': {
        …
    }
    …
}
```

From talks with @ttmc, @sbellem, @libscott.

### PR DESCRIPTION
**Pull request by libscott**
Implementation of #828. ~~This hasn't been generally approved yet.~~ This has been generally approved.

Still to do, if/when given green light:

- [x] Docs changes

And separately: 

- Corresponding PR in bigchaindb-driver
- Ditto bdb-transaction-cli

Fixes #828

### CODE REVIEW

File: bigchaindb/db/backends/rethinkdb.py [hunk 1]
```diff
@@ -202,10 +202,9 @@ def get_asset_by_id(self, asset_id):
              .get_all(asset_id, index='asset_id')
              .concat_map(lambda block: block['block']['transactions'])
              .filter(lambda transaction:
-                     transaction['transaction']['asset']['id'] == asset_id)
+                     transaction['asset']['id'] == asset_id)
              .filter(lambda transaction:
-                     transaction['transaction']['operation'] == 'CREATE')
-             .pluck({'transaction': 'asset'}))
```
Comment (r-marques) [inline bigchaindb/db/backends/rethinkdb.py]: Why was this removed?

Comment (libscott) [inline bigchaindb/db/backends/rethinkdb.py]: Hmm, looks odd indeed! Will investigate...

Comment (r-marques) [inline bigchaindb/db/backends/rethinkdb.py]: everything will work the same with or without pluck because the structure of the document is maintained. The only difference is that all other keys are empty
